lib/ui/text.dart

```diff
@@ -1684,6 +1863,13 @@ class Paragraph extends NativeFieldWrapperClass2 {
   // in the C++ code. If we straighten out the C++ dependencies, we can remove
   // this indirection.
   void _paint(Canvas canvas, double x, double y) native 'Paragraph_paint';
+
+  /// Returns the full list of [LineMetrics] that describe in detail the various
+  /// metrics of each laid out line.
+  ///
+  /// This can potentially return a large amount of data, so it is not recommended
+  /// to repeatedly call this. Instead, cache the results.
```

Comment:
Is this value available prior to calling `layout`? If not, let's document that.

Does the value depend on `ParagraphConstraints` passed to the `layout` method? If not, we could cache right here and not scare the user with the large amount of data disclaimer. Actually, even if it does depend on the `ParagraphConstratins`, we can still cache and invalidate the cache next time `layout` is called.